pkg/topology/kademlia/kademlia.go

```diff
@@ -859,13 +828,8 @@ func (k *Kad) Disconnected(peer p2p.Peer) {
 	po := swarm.Proximity(k.base.Bytes(), peer.Address.Bytes())
 	k.connectedPeers.Remove(peer.Address, po)
 
-	k.waitNextMu.Lock()
-	newInfo := retryInfo{tryAfter: time.Now().Add(timeToRetry), failedAttempts: 0}
-	if info, ok := k.waitNext[peer.Address.ByteString()]; ok {
-		newInfo.failedAttempts = info.failedAttempts
-	}
-	k.waitNext[peer.Address.ByteString()] = newInfo
-	k.waitNextMu.Unlock()
+	failedAttempts := k.waitNext.Attempts(peer.Address)
+	k.waitNext.Set(peer.Address, time.Now().Add(timeToRetry), failedAttempts)
```

Comment:
This is actually, not an atomic operation. I would suggest to create a new method on WaitNext just to set the time and leave the counter as it is under a single lock.